baseapp/baseapp_test.go

```diff
@@ -821,6 +841,9 @@ func TestDeliverTx(t *testing.T) {
 
 			res := app.DeliverTx(abci.RequestDeliverTx{Tx: txBytes})
 			require.True(t, res.IsOK(), fmt.Sprintf("%v", res))
+			events := res.GetEvents()
+			require.Len(t, events, 2, "should contain message type and counter events")
+			require.Equal(t, counterEvent(counter).ToABCIEvents()[0], events[1], "update counter event")
```

Comment:
Are we testing that the resulting events contain events from both the message handler _and_ the AnteHandler?

Reply:
I think this needs a counterEvent call on like 679